thrust/thrust/system/tbb/detail/reduce_by_key.inl

```diff
@@ -58,21 +59,17 @@ inline L divide_ri(const L x, const R y)
   return (x + (y - 1)) / y;
 }
 
-template <typename InputIterator, typename BinaryFunction, typename OutputIterator = void>
+template <typename InputIterator, typename BinaryFunction, typename SFINAE = void>
 struct partial_sum_type
-    : thrust::detail::eval_if<thrust::detail::has_result_type<BinaryFunction>::value,
-                              thrust::detail::result_type<BinaryFunction>,
-                              thrust::detail::eval_if<thrust::detail::is_output_iterator<OutputIterator>::value,
-                                                      thrust::iterator_value<InputIterator>,
-                                                      thrust::iterator_value<OutputIterator>>>
-{};
```

Comment:
The primary template was never used, since `partial_sum_type` is always instantiated with two types instead and never with three.